### ISSUE TITLE
UTF-8 unusable on windows with 1.14.4 - 1.14.5

### ISSUE THREAD

**Issue by hmaarrfk** [conda-forge/hdf5-feedstock#240]
### Solution to issue cannot be found in the documentation.

- [x] I checked the documentation.

### Issue

See https://github.com/HDFGroup/hdf5/issues/5037 for more details but basically:

1. A patch in 1.14.4 broke UTF-8 filename support on windows.
2. This is blocking h5py from being compiled on windows.

I personally think UTF-8 is pretty important. I've seen users assume they can put any character they want (commas, quotes, colons, you name it.....) in their file names.

However, today HDF5 is splitting the conda ecosystem making it more difficult to co-install packages.

I think there are a few options for conda-forge:

1. Downgrade to 1.14.3
2. Dual build: 1.14.3 + 1.14.4 (heavy maintenance downstream)
3. Ship 1.14.4 and say it is a known issue ([h5py's pypi package might do this](https://github.com/HDFGroup/hdf5/issues/5037#issuecomment-2456741153))
4. Revert the offending patch on our version 1.14.4 https://github.com/HDFGroup/hdf5/pull/4172

I'm not sure what other maintainers have in mind, but it would be good to get to a decision and to unblock this by the end of the week.

### Installed packages

hdf5 1.14.4 and 1.14.5

### Environment info

Windows.

**Comment by hmaarrfk** [comment]
I would like to move forward with Option 2:

* Dual build 1.14.3 and 1.14.4/5/6

while this gets resolved upstream.

This isn't too different than what we've done in the past with HDF5 updates. HDF5 updates often take time, and the libraries that consume them can be quite old (due to the strong history of HDF5) and makes it difficult for packages to migrate.

I feel like 1.14.4/5 is actually quite disruptive to the ecosystem as a whole, while the UTF-8 issue isn't resolved.

Thoughts are welcome.

I've rebuilt 1.14.3 with the new features that have been contributed.

If an other maintainer agrees I have a checklist of the next steps outlined in https://github.com/conda-forge/hdf5-feedstock/pull/241

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

so as to add a sacred branch

**Comment by minrk** [comment]
Following from https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/6900

As a maintainer of several of the heavy matrix feedstocks, the root of most of them is petsc, which I [merged yesterday](https://github.com/conda-forge/petsc-feedstock/pull/224), launching the several really heavy ones (which add a Python matrix on top of the mpi + petsc scalar). I'm happy to close those PRs and revert the petsc PR, if that's what we should do. But it's not quite clear to me what action to take, other than don't merge the PRs the closed migrator already issued.

A bit tangential, but I'm also totally open to input to trim the matrix on several of the heavy petsc dependents, but I don't know how to make those decisions. ppc is by far the most costly platform to support (both in terms of compute and maintainer time), and I am not aware of a single user. It appears to be ~entirely unused in this stack of packages, except for similarly unused ppc builds of downstream packages, at least from anaconda.org download counts. I don't know who the conda-forge ppc builds are for, so I don't know if/when it's appropriate to drop ppc support.

**Comment by h-vetinari** [comment]
> I'm happy to close those PRs and revert the petsc PR, if that's what we should do. But it's not quite clear to me what action to take, other than don't merge the PRs the closed migrator already issued.

You don't need to revert PRs, just ensure that the next respective PR rerenders the feedstock correctly (which will remove half the matrix; the ones for 1.14.4).

> ppc is by far the most costly platform to support (both in terms of compute and maintainer time), and I am not aware of a single user.

You're free to drop it from your feedstocks if it causes maintenance issues, but for the larger picture, this is a discussion for https://github.com/conda-forge/conda-forge.github.io I think. I agree that whenever I looked at download counts, PPC is at least an order of magnitude behind the second-least-used platform, so I can see where you're coming from.

<details>
<summary>example</summary>

[openssl 3.4.0](https://anaconda.org/conda-forge/openssl/files)

| artefact | uptime | downloads |
| -- | -- | -- |
|               linux-64/openssl-3.4.0-hb9d3cd8_0.conda | 1 month and 26 days ago | 4'039'394 |
|               linux-aarch64/openssl-3.4.0-h86ecc28_0.conda | 1 month and 26 days ago | 134'917 |
|               linux-ppc64le/openssl-3.4.0-h190368a_0.conda | 1 month and 26 days ago | 10'885 |
|               osx-64/openssl-3.4.0-hd471939_0.conda | 1 month and 26 days ago | 186'830 |
|               osx-arm64/openssl-3.4.0-h39f12f2_0.conda | 1 month and 26 days ago | 215'090 |
|               win-64/openssl-3.4.0-h2466b09_0.conda | 1 month and 26 days ago | 507'144 |

OTOH, 10k users isn't nothing either 🤷 

</details>

**Comment by h-vetinari** [comment]
Did we ever start a migration for 1.14.6?

**Comment by hmaarrfk** [comment]
Yeah sorry for not referencing the reason. it was a stressful time with this UTF-8 stuff.

There is also https://github.com/conda-forge/hdf5-feedstock/pull/254 that is ready but i want to see if we can do the cmake transition too!
